Clarify semantics: Deprecate and add fields and codelists, update descriptions (#26)
open-contracting-extensions/ocds_finance_extension: README.md

## Codelists

> The `financingPartyType.csv` codelist is based on the list on [Page 57 of the World Bank PPP Disclosure Framework](https://thedocs.worldbank.org/en/doc/143671469558797229-0100022016/original/FrameworkPPPDisclosure071416.pdf#page=57).

Reviewer: We've diverged so much, let's just delete this section.

Author: Done in https://github.com/open-contracting-extensions/ocds_finance_extension/pull/26/commits/73c7856e8130bd1dcdb636823d5b55b2204b2e3c

{
  "planning": {
    "budget": {
      "finance": [
        {
          "id": "1",
          "financingParty": {
            "id": "1",
            "name": "Development Bank of South Africa"
          },

>           "source": "Green Climate Fund",
>           "financingPartyType": "bilateral",

Reviewer: Easier to read with two financingParty... fields next to each other.

```suggestion
          "financingPartyType": "bilateral",
          "source": "Green Climate Fund",
```

## Changelog

### 2023-11-13

> * Clarify semantics:

Reviewer: ```suggestion
* Replace fields, to clarify semantics:
```

>   * Replace Finance.financeType (financeType.csv) with `Finance.financingPartyType` (`financingPartyType.csv`) and `Finance.type` (`financingArrangementType.csv`). Notably:
>     * "publicBondIssue" is replaced by 'debt' in `financingArrangementType.csv` and 'bond' `assetClass.csv`
>     * "supplierCredit" is replaced by 'vendor' in the `financingPartyType.csv` codelist
>   * Replace Finance.financeCategory (financeCategory.csv) with `Finance.assetClass` (`assetClass.csv`), `Finance.type` (`financingArrangementType.csv`) and `Finance.repaymentPriority` (`debtRepaymentPriority.csv`). Notably:
>     * "equity" is replaced by 'equity' in `assetClass`
>     * "seniorDebt" is replaced by 'debt' in `assetClass.csv` and 'senior' in `debtRepaymentPriority.csv`
>     * "mezzanineDebt" is replaced by 'debt' and 'equity' in `assetClass.csv` and 'subordinated' in `debtRepaymentPriority.csv`
>     * "grant" is replaced by 'grant' in `financingArrangementType.csv`
>     * "guarantee" is replaced by 'guarantee' in `financingArrangementType.csv`

Reviewer: I added exceptions for the removed codes.

```suggestion
  * Replace `Finance.financeType` (`financeType.csv`) with `Finance.financingPartyType` (`financingPartyType.csv`) and `Finance.type` (`financingArrangementType.csv`). Notably:
    * 'publicBondIssue' is replaced by 'debt' in `assetClass.csv` and 'bond' in `financingArrangementType.csv`
    * 'supplierCredit' is replaced by 'vendor' in `financingPartyType.csv`
  * Replace `Finance.financeCategory` (`financeCategory.csv`) with `Finance.assetClass` (`assetClass.csv`), `Finance.type` (`financingArrangementType.csv`) and `Finance.repaymentPriority` (`debtRepaymentPriority.csv`). Notably:
    * 'equity' is replaced by 'equity' in `assetClass.csv`
    * 'seniorDebt' is replaced by 'debt' in `assetClass.csv` and 'senior' in `debtRepaymentPriority.csv`
    * 'mezzanineDebt' is replaced by 'debt' and 'equity' in `assetClass.csv` and 'subordinated' in `debtRepaymentPriority.csv`
    * 'grant' is replaced by 'grant' in `financingArrangementType.csv`
    * 'guarantee' is replaced by 'guarantee' in `financingArrangementType.csv`
```

Author: Looks like the lint check is still failing on the old codes.

Reviewer: Should work now

* Add fields:

>   * `Finance.resultsBased`

Reviewer: Use alphabetical order

Author: Done in https://github.com/open-contracting-extensions/ocds_finance_extension/pull/26/commits/73c7856e8130bd1dcdb636823d5b55b2204b2e3c

  * `Finance.concessional`
  * `Finance.source`
  * `Finance.repaymentPeriod`

> * Update descriptions:

Reviewer: ```suggestion
* Update descriptions, for clarity:
```

### 2020-04-17

* Add `planning.budget.finance` field.

> * Fix description of financeCategory.

Reviewer: ```suggestion
* Fix description of `Finance.financeCategory`.
```
